rpcs3/Emu/Cell/Modules/cellAudio.cpp

```diff
@@ -569,57 +561,90 @@ namespace audio
 				raw.buffering_enabled != new_raw.buffering_enabled ||
 				raw.time_stretching_threshold != new_raw.time_stretching_threshold ||
 				raw.enable_time_stretching != new_raw.enable_time_stretching ||
-				raw.convert_to_u16 != new_raw.convert_to_u16 ||
-				raw.start_threshold != new_raw.start_threshold ||
-				raw.sampling_period_multiplier != new_raw.sampling_period_multiplier ||
+				raw.convert_to_s16 != new_raw.convert_to_s16 ||
 				raw.downmix != new_raw.downmix ||
 				raw.renderer != new_raw.renderer)
 			{
 				g_audio.cfg.raw = new_raw;
-				g_audio.m_update_configuration = true;
+				g_audio.m_update_configuration = raw.renderer != new_raw.renderer ? audio_backend_update::ALL : audio_backend_update::PARAM;
 			}
 		}
 	}
 }
 
-void cell_audio_thread::update_config()
+void cell_audio_thread::update_config(bool backend_changed)
 {
 	std::lock_guard lock(mutex);
 
 	// Clear ringbuffer
 	ringbuffer.reset();
 
 	// Reload config
-	cfg.reset();
+	cfg.reset(backend_changed);
 
 	// Allocate ringbuffer
 	ringbuffer.reset(new audio_ringbuffer(cfg));
+
+	// Reset thread state
+	reset_counters();
+}
+
+void cell_audio_thread::reset_counters()
+{
+	m_counter = 0;
+	m_start_time = ringbuffer->get_timestamp();
+	m_last_period_end = m_start_time;
+	m_dynamic_period = 0;
+	m_backend_failed = false;
 }
 
 void cell_audio_thread::operator()()
 {
+	if (cfg.raw.provider != audio_provider::cell_audio)
+	{
+		return;
+	}
+
 	thread_ctrl::scoped_priority high_prio(+1);
 
+	// Init audio config
+	cfg.reset();
+
 	// Allocate ringbuffer
 	ringbuffer.reset(new audio_ringbuffer(cfg));
 
 	// Initialize loop variables
-	m_counter = 0;
-	m_start_time = ringbuffer->get_timestamp();
-	m_last_period_end = m_start_time;
-	m_dynamic_period = 0;
+	reset_counters();
 
 	u32 untouched_expected = 0;
-	//u32 in_progress_expected = 0;
 
 	// Main cellAudio loop
 	while (thread_ctrl::state() != thread_state::aborting)
 	{
-		if (m_update_configuration)
+		const auto update_req = m_update_configuration.observe();
+		if (update_req != audio_backend_update::NONE)
 		{
 			cellAudio.warning("Updating cell_audio_thread configuration");
-			update_config();
-			m_update_configuration = false;
+			update_config(update_req == audio_backend_update::ALL);
+			m_update_configuration = audio_backend_update::NONE;
+		}
+
+		if (!ringbuffer->get_operational_status())
+		{
+			if (m_backend_failed)
+			{
+				thread_ctrl::wait_for(500 * 1000);
+			}
+
+			cellAudio.error("Backend stopped unexpectedly (likely device change). Attempting to recover...");
```

Comment:
this should be logged first.
Also, this will now always happen if you switch devices with XAudio

Reply:
This is intended behavior.